spinner/eglsurface.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © 2019 UBports Foundation.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <EGL/egl.h>
+
+#include <memory>
+#include <functional>
+
+class EglSurface
```

Comment:
Maybe rename to `SpinnerEglSurface`, to avoid confusion with `EGLSurface` (actual type from EGL).